<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2017-07-17" />

<title>Depletion</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Logoplot dash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Depletion</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2017-07-17</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-07-18</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 49915de</p>
<!-- Add your analysis here -->
<p>We have seen some negative logo plots and depletion exmaples in previous vignettr. Now, We are interested in finding the depletion of one particular base(e.g. (0,1/3,1/3,1/3)) and then study whether the corresponding transcription factors have special functionality.</p>
<p>For convenience, I call this kind of case as “perfect depletion”. Perfect depletion corresponds to <span class="math inline">\(p=(0,\frac{1}{3},\frac{1}{3},\frac{1}{3})\)</span> in position weight matrix(PWM) where 0 is the weight of depleted base. In such case, the information content(IC) is then <span class="math inline">\((0.415,0,0,0)\)</span>, where <span class="math inline">\(0.415\)</span> is the negative IC. With pseudocounts, we would not have the case like <span class="math inline">\(p\)</span>. So now suppose the perfect depletion case is <span class="math inline">\(q=(0.01,0.33,0.33,0.33)\)</span>, then the IC is <span class="math inline">\((0.35,0,0,0)\)</span>. One simple way to find such cases is to measure the difference between the largest IC on nagative part and the second largest IC on each position. The difference is denoted as <span class="math inline">\(d\)</span>. Then if the difference is larger than a threshold <span class="math inline">\(\alpha\)</span>, the perfect depletion exists.</p>
<p>Here, <span class="math inline">\(\alpha\)</span> is chosen empirically. For a database(HOCOMOCO, JASPAR 2014), suppose it has total <span class="math inline">\(m\)</span> PWMs and take the largest <span class="math inline">\(d\)</span> for each PWMs, then we have <span class="math inline">\(m\)</span> differences. Sort the <span class="math inline">\(m\)</span> number descendingly and find the smallest <span class="math inline">\(d\)</span> such that the IUPAC code corresponding to that PWM has either B,D,H or V and after that <span class="math inline">\(d\)</span>, there are seldomly such PWMs.(See <a href="http://www.bioinformatics.org/sms/iupac.html" class="uri">http://www.bioinformatics.org/sms/iupac.html</a> for the IUPAC code.) Then the value this smallest <span class="math inline">\(d\)</span> is the threshold <span class="math inline">\(\alpha\)</span>. The rationale is that in IUPAC code, B means C or Gã€€or T, which actually corresponds to the depletion in A. The same reason holds for D, H and V.</p>
<p>Also, if the sum of counts for each position in pfm is smaller than 20, the corresponding pfm is discarded.(One alternative way is to use <em>dash</em>)</p>
<p>For transcription factor with perfect depletion, we record its name, depleted positions, bases, distances, PWM and the neighbors of depleted bases.</p>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<p>Firstly, we get the depletion list of each database. There are some overlaps of JASPAR 2014 and HOCOMOCO. So here our analysis of depleted TFs focuses on HOCOMOCO database, which consists of Human and Mouse data.</p>
<pre class="r"><code>#get the list of perfect depeltion from JASPAR, HOCOMOCO
DepJaspar=depletion(PWMJaspar,findalpha(scJaspar,50,PWMJaspar))
DepHuman=depletion(PWMHuman,findalpha(scHuman,50,PWMHuman))
DepMouse=depletion(PWMMouse,findalpha(scMouse,50,PWMMouse))
paste(&#39;There are&#39;,length(DepHuman),&#39;perfect depletion in Human data&#39;)</code></pre>
<pre><code>[1] &quot;There are 41 perfect depletion in Human data&quot;</code></pre>
<pre class="r"><code>paste(&#39;There are&#39;,length(DepMouse),&#39;perfect depletion in Mouse data&#39;)</code></pre>
<pre><code>[1] &quot;There are 33 perfect depletion in Mouse data&quot;</code></pre>
<p>We then analyzed the genes that produce these TFs(with delpetion), by the censusPathDB. The idea is that each TF is produced by a gene and then we test if the gene producing TF with depletion is related to a specific function. It turns out that the depletion is probably not connected with specific functional behavior. See <a href="depletion_tfsea.html">here</a> for details.</p>
<p>From the analysis of HOCOMOCO plots, we found that two TF families are of interests - SOX and IRF, the reason is that there are several Trancription factor with perfect delpeltion, namely SOX17, SOX3 and IRF3,IRF4,IRF8. Below are the logo plots of Mouse IRF family.</p>
<pre class="r"><code>for (i in grep(&#39;IRF&#39;,names(PWMMouse))){
  if(sum(is.na(PWMMouse[[i]]))==0){
    print(GetIUPACSequence(PWMMouse[[i]]))
  nlogomaker(PWMMouse[[i]],logoheight = &#39;ic&#39;,color_profile = color_profile,frame_width = 1, pop_name = names(PWMMouse[i]),ylimit=2.1)
}}</code></pre>
<pre><code>[1] &quot;RAAASYGAAASY&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;SGAAAGYGAAASYR&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;GGAAASYGAAASBRRRA&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;AAAARAGRAAVTGARA&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;GAAASYRAAW&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;GRGRAAVTGAAASYR&quot;</code></pre>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>One kind of plots that are of particular interests is the dimer. For example, the logo plots of IRF families are dimers, with depletion in the middle and enrichment in the two sides. We found that dimer is common and usually has the characteristic that depletion gap is between two palindromic sequences(or same sequence like IRF, not common). For example, TGGGG__CCCCA.</p>
</div>
<div id="depletion-frequncies" class="section level1">
<h1>Depletion frequncies</h1>
<p>This section presents the delpletion frequency for A,G,C,T at different postions. For each base, in the PWM of a transcription factor, suppose the depletion occurs at the <span class="math inline">\(n_{base}^{th}\)</span> postion and the total number of postion is <span class="math inline">\(n\)</span>. Then we measure the relative distance of postion to the center as: <span class="math inline">\(d=|\frac{n_{base}}{n}-0.5|\)</span>. If d is close to 0.5, then the depletion position is close to the end. If d is close to 0, then the delpetion position is close to the center.</p>
<p><img src="figure/Depletion.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Depletion.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Depletion.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Depletion.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>From the histograms, there are more depletion at the ends than that at the center for the bases.</p>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 15063)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] Logolas_1.1.2       TFBSTools_1.14.0    JASPAR2014_1.12.0  
[4] Biostrings_2.43.8   XVector_0.15.2      IRanges_2.9.19     
[7] S4Vectors_0.13.17   BiocGenerics_0.22.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.11                lattice_0.20-35            
 [3] GO.db_3.4.1                 png_0.1-7                  
 [5] Rsamtools_1.27.16           gtools_3.5.0               
 [7] rprojroot_1.2               digest_0.6.12              
 [9] R6_2.2.0                    GenomeInfoDb_1.12.0        
[11] plyr_1.8.4                  backports_1.0.5            
[13] RSQLite_1.1-2               evaluate_0.10              
[15] httr_1.2.1                  ggplot2_2.2.1              
[17] zlibbioc_1.21.0             lazyeval_0.2.0             
[19] annotate_1.54.0             R.utils_2.5.0              
[21] R.oo_1.21.0                 Matrix_1.2-9               
[23] rmarkdown_1.6               splines_3.4.0              
[25] BiocParallel_1.10.0         readr_1.1.1                
[27] stringr_1.2.0               CNEr_1.12.0                
[29] RCurl_1.95-4.8              munsell_0.4.3              
[31] DelayedArray_0.2.0          compiler_3.4.0             
[33] rtracklayer_1.35.12         seqLogo_1.42.0             
[35] DirichletMultinomial_1.18.0 htmltools_0.3.5            
[37] KEGGREST_1.16.0             SummarizedExperiment_1.6.0 
[39] tibble_1.3.0                GenomeInfoDbData_0.99.0    
[41] matrixStats_0.52.2          XML_3.98-1.6               
[43] TFMPvalue_0.0.6             GenomicAlignments_1.11.12  
[45] bitops_1.0-6                R.methodsS3_1.7.1          
[47] grid_3.4.0                  xtable_1.8-2               
[49] gtable_0.2.0                DBI_0.6-1                  
[51] git2r_0.18.0                magrittr_1.5               
[53] scales_0.4.1                stringi_1.1.5              
[55] reshape2_1.4.2              RColorBrewer_1.1-2         
[57] tools_3.4.0                 BSgenome_1.44.0            
[59] Biobase_2.35.1              poweRlaw_0.70.0            
[61] hms_0.3                     yaml_2.1.14                
[63] AnnotationDbi_1.38.1        colorspace_1.3-2           
[65] GenomicRanges_1.27.23       caTools_1.17.1             
[67] memoise_1.1.0               VGAM_1.0-3                 
[69] knitr_1.15.1               </code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
