<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2017-08-22" />

<title>First Look at Logolas</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Logoplot dash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="tutorial.html">Tutorial</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">First Look at Logolas</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2017-08-22</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-08-24</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 46898f9</p>
<!-- Add your analysis here -->
<div id="first-look-at-logolas" class="section level1">
<h1>First look at Logolas</h1>
<div id="logo-plots" class="section level2">
<h2>logo plots</h2>
<p>One of the most basic applications of logo plots is to visualize the DNA sequence alignment(motif), which consists of four letters(A,C,G and T), corresponding to the four nucleotides(Adenine, Cytosine, Guanine, Thymine). The function <code>logomaker()</code> in Logolas provides the standard logo plots with the position weight matrix(PWM) as input. Here, we start with the same demo example provided in the <a href="https://bioconductor.org/packages/release/bioc/html/seqLogo.html">seqLogo</a> vignette.</p>
<pre class="r"><code>library(Logolas)
mFile &lt;- system.file(&quot;Exfiles/pwm1&quot;, package=&quot;seqLogo&quot;)
m &lt;- read.table(mFile)
p &lt;- seqLogo::makePWM(m)
color_profile &lt;- list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(p@pwm)[1],name =&quot;Spectral&quot;))
logomaker(p@pwm,color_profile = color_profile, frame_width = 1,ic.scale = TRUE)</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As a comparison, we present the logo plot of the same PWM from the package <code>seqLogo</code>.</p>
<pre class="r"><code>seqLogo::seqLogo(p)</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>User could specify the color for each letter through the <code>color_profile</code>.</p>
<p>If <code>ic.scale</code>= TRUE, the heights of the bars at each position are determined by the Shannon entropy. The information content for position <span class="math inline">\(i\)</span> is <span class="math inline">\(IC_i=-\Sigma_b p_{b}\times \log_2p_{b}-(-\Sigma_b q_{b,i}\times \log_2q_{b,i})\)</span>, where <span class="math inline">\(q_{b,i}\)</span> is the relative frequency of base <span class="math inline">\(b\)</span> at position <span class="math inline">\(i\)</span> and <span class="math inline">\(p_{b}\)</span> is the background probability of base <span class="math inline">\(b\)</span>. By default, the background probability of the each base is 0.25. So the information content at position <span class="math inline">\(i\)</span> is <span class="math inline">\(IC_i=\Sigma_bq_{b,i}\times \log_2(q_{b,i}/0.25)\)</span>.</p>
<p><code>Logolas</code> could also combine multiple plots into one overall graph. The figure below demonstrates the multiple panel plot as well as some variants of logo plots.</p>
<pre class="r"><code>library(grid)

grid.newpage()
layout.rows &lt;- 2
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(5,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(rep(5,layout.rows), rep(&quot;null&quot;, 1))))
plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
#change the color of letters
#change the diverging palettes
color_profile1=list(&quot;type&quot; = &quot;per_row&quot;,
&quot;col&quot; = RColorBrewer::brewer.pal(dim(p@pwm)[1],name =&quot;PiYG&quot;))
seekViewport(paste0(&quot;plotlogo&quot;, 1))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile1,
          frame_width = 1,
          newpage = FALSE,
          pop_name = &#39;Change color&#39;,
          control = list(viewport.margin.left = 5))

#change the y scale:
#if yscale_change = FALSE, then the height of y axis would be 2. 
seekViewport(paste0(&quot;plotlogo&quot;, 2))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile,
          frame_width = 1,
          newpage = FALSE,
          pop_name = &#39;change y scale&#39;,
          yscale_change = FALSE,
          control = list(viewport.margin.left = 5))
#Normalize the height of bars to 1
seekViewport(paste0(&quot;plotlogo&quot;, 3))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile,
          frame_width = 1,
          newpage = FALSE,
          ic.scale = FALSE,
          pop_name = &#39;Normalize the height&#39;,
          control = list(viewport.margin.left = 5))
#change the background probability
#And modify the title and the axis label 
seekViewport(paste0(&quot;plotlogo&quot;, 4))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile,
          frame_width = 1,
          bg=c(0.32, 0.18, 0.2, 0.3),
          newpage = FALSE,
          pop_name = &#39;Change background prob&#39;,
          control = list(viewport.margin.left = 5))</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="choose-the-entropy" class="section level2">
<h2>Choose the entropy</h2>
<p>Besides the default Shannon entropy, <code>logomaker</code> could also get the information content using Renyi entropy. The information content at position <span class="math inline">\(i\)</span> is <span class="math inline">\(IC_{i,\alpha}=\frac{1}{1-\alpha}\Sigma_b\log_2(q_{b,i}^\alpha-0.25^{1-\alpha})\)</span>. When <span class="math inline">\(\alpha\rightarrow1\)</span>, the limiting value of Renyi entropy is the Shannon entropy. The figure below shows the logo plot with different values of <span class="math inline">\(\alpha\)</span>.</p>
<pre class="r"><code>grid.newpage()
layout.rows &lt;- 1
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(6,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(c(20,50), rep(&quot;lines&quot;, 2))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile,
          frame_width = 1,
          newpage = FALSE,
          pop_name = &#39;Logo plot, Alpha=2&#39;,
          control = list(viewport.margin.left = 5,alpha=2))


seekViewport(paste0(&quot;plotlogo&quot;, 2))
logomaker(p@pwm,xlab = &#39;position&#39;,color_profile = color_profile,
          frame_width = 1,
          newpage = FALSE,
          pop_name = &#39;Logo plot, Alpha=0.2&#39;,
          control = list(viewport.margin.left = 5,alpha=0.2))</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="negative-logo-plots" class="section level2">
<h2>Negative logo plots</h2>
<p>In the logo plot, the enrichment is highlighted while the depletion is usually overlooked. We hence propose the negative logo plot, in which the depletion is plotted down below the zero on y-axis so that we can find both strong positive and negative effects.</p>
<p>The <code>nlogomaker</code> function provides negative logo plot. And there are three different options of logo heights: information content, log probability and log-odds.</p>
<p>The negative logo plot actually shows the relative change of base frequencies. To achieve this, the median of base frequencies is subtracted at a position. For example, the relative frequencies at a position is <span class="math inline">\(P=(0.7,0.15,0.15,0)\)</span>; subtracted by the median, it becomes <span class="math inline">\(P_{neg}=(0.55,0,0,-0.15)\)</span>. Then the total heights are allocated to the positive and negative parts and further allocated to each base, based on <span class="math inline">\(P_{neg}\)</span>.</p>
<p>Below we show the negative logo plot using log and log-odds heights for the same example as above.</p>
<pre class="r"><code>grid.newpage()
layout.rows &lt;- 1
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(6,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(c(20,50), rep(&quot;lines&quot;, 2))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
nlogomaker(p@pwm,logoheight = &#39;log&#39;,color_profile = color_profile,frame_width = 1,newpage = FALSE)


seekViewport(paste0(&quot;plotlogo&quot;, 2))
nlogomaker(p@pwm,logoheight = &#39;log_odds&#39;,color_profile = color_profile,frame_width = 1,newpage = FALSE)</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As seen from the above figures, the negative logo plot presents both the enrichment and depletion. For example, there is an obvious depletion of T at the position 4. It may be considered that the transcription factor(TF) would find both enrichment and depletion to decide whether to bind. For example, at position 4, the depletion of T probably means the TF dislikes T at that position.</p>
</div>
<div id="depletion-weights" class="section level2">
<h2>Depletion weights</h2>
<p>Logolas further allow users to customize the weight of depletion. As the depletion weight increasing, the depletion would look more prominent. The default depletion weight is 0.5 and comparisons of different depletion weights are shown below. The weights chosen are 0.1,0.3,0.5(default),1.</p>
<pre class="r"><code>grid.newpage()
layout.rows &lt;- 2
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(5,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(rep(5,layout.rows), rep(&quot;null&quot;, 1))))
plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)
seekViewport(paste0(&quot;plotlogo&quot;, 1))
nlogomaker(p@pwm,
           logoheight = &#39;log_odds&#39;,
           color_profile = color_profile,
           frame_width = 1,
           newpage = FALSE,
           pop_name = &#39;Weight=0.1&#39;,
           control = list(depletion_weight=0.1))


seekViewport(paste0(&quot;plotlogo&quot;, 2))
nlogomaker(p@pwm,
           logoheight = &#39;log_odds&#39;,
           color_profile = color_profile,
           frame_width = 1,
           newpage = FALSE,
            pop_name = &#39;Weight=0.3&#39;,
           control = list(depletion_weight=0.3))

seekViewport(paste0(&quot;plotlogo&quot;, 3))
nlogomaker(p@pwm,
           logoheight = &#39;log_odds&#39;,
           color_profile = color_profile,
           frame_width = 1,
           newpage = FALSE,
            pop_name = &#39;Weight=0.5&#39;,
           control = list(depletion_weight=0.5))

seekViewport(paste0(&quot;plotlogo&quot;, 4))
nlogomaker(p@pwm,
           logoheight = &#39;log_odds&#39;,
           color_profile = color_profile,
           frame_width = 1,
           newpage = FALSE,
            pop_name = &#39;Weight=1&#39;,
           control = list(depletion_weight=1))</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="border-and-fill" class="section level2">
<h2>Border and fill</h2>
<p>User could customize the border and fill of the logos. For example, the user may choose to fill the enrichment while only show the border of depletion. The figure below shows the various option on making the logos.</p>
<pre class="r"><code>grid.newpage()
layout.rows &lt;- 2
layout.cols &lt;- 2
top.vp &lt;- viewport(layout=grid.layout(layout.rows, layout.cols,
                                      widths=unit(rep(5,layout.cols), rep(&quot;null&quot;, 2)),
                                      heights=unit(rep(5,layout.rows), rep(&quot;null&quot;, 1))))

plot_reg &lt;- vpList()
l &lt;- 1
for(i in 1:layout.rows){
  for(j in 1:layout.cols){
    plot_reg[[l]] &lt;- viewport(layout.pos.col = j, layout.pos.row = i, name = paste0(&quot;plotlogo&quot;, l))
    l &lt;- l+1
  }
}


plot_tree &lt;- vpTree(top.vp, plot_reg)

pushViewport(plot_tree)

seekViewport(paste0(&quot;plotlogo&quot;, 1))
nlogomaker(p@pwm,xlab = &#39;position&#39;,logoheight = &quot;log&quot;,
           color_profile = color_profile,
           bg = c(0.25, 0.25, 0.25, 0.25),
           frame_width = 1,
           control = list(tofill_pos = TRUE, tofill_neg=FALSE,
                          logscale = 0.2, quant = 0.5,
                          depletion_weight = 0.5),
           newpage = FALSE)

seekViewport(paste0(&quot;plotlogo&quot;, 2))
nlogomaker(p@pwm,xlab = &#39;position&#39;,logoheight = &quot;log&quot;,
           color_profile = color_profile,
           bg = c(0.25, 0.25, 0.25, 0.25),
           frame_width = 1,
           control = list(tofill_pos = FALSE, tofill_neg=TRUE,
                          logscale = 0.2, quant = 0.5,
                          depletion_weight = 0.5),
           newpage = FALSE)

seekViewport(paste0(&quot;plotlogo&quot;, 3))
nlogomaker(p@pwm,xlab = &#39;position&#39;,logoheight = &quot;log&quot;,
           color_profile = color_profile,
           bg = c(0.25, 0.25, 0.25, 0.25),
           frame_width = 1,
           control = list(tofill_pos = TRUE, tofill_neg=TRUE,
                          logscale = 0.2, quant = 0.5,
                          depletion_weight = 0.5),
           newpage = FALSE)

seekViewport(paste0(&quot;plotlogo&quot;, 4))
nlogomaker(p@pwm,xlab = &#39;position&#39;,logoheight = &quot;log&quot;,
           color_profile = color_profile,
           bg = c(0.25, 0.25, 0.25, 0.25),
           frame_width = 1,
           control = list(tofill_pos = FALSE, tofill_neg=FALSE,
                          logscale = 0.2, quant = 0.5,
                          depletion_weight = 0.5),
           newpage = FALSE)</code></pre>
<p><img src="figure/FirstLook.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 15063)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] Logolas_1.1.2

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.12       digest_0.6.12      rprojroot_1.2     
 [4] backports_1.0.5    stats4_3.4.0       git2r_0.18.0      
 [7] magrittr_1.5       evaluate_0.10      stringi_1.1.5     
[10] rmarkdown_1.6      RColorBrewer_1.1-2 tools_3.4.0       
[13] stringr_1.2.0      yaml_2.1.14        compiler_3.4.0    
[16] seqLogo_1.42.0     htmltools_0.3.5    knitr_1.15.1      </code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
