<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Dongyue Xie" />

<meta name="date" content="2017-07-17" />

<title>Negative Logo plot</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Logoplot dash</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Negative Logo plot</h1>
<h4 class="author"><em>Dongyue Xie</em></h4>
<h4 class="date"><em>2017-07-17</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2017-07-18</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 49915de</p>
<!-- Add your analysis here -->
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In the logo plots, the transcription factor binding sequence patterns are visualized. The enrichment is highlighted while the depletion is usually overlooked. In the negative logo plot, the depletion is plotted down below the zero on y-axis so that we can find both strong positive and negative effects.</p>
<p>The <em>nlogomaker</em> function from R package “Logolas” provides negative logo plots. And there are three different options of logo heights: information content, log probability and log-odds. More details can be found here: <a href="https://github.com/kkdey/Logolas" class="uri">https://github.com/kkdey/Logolas</a>.</p>
<p>The following section presents some negative logo plots to give us an initial look at them.</p>
</div>
<div id="example-plots" class="section level1">
<h1>Example plots</h1>
<p>The data of this example are from JASPAR2014. JASPAR is a collection of transcription factor DNA-binding preferences. We used the data from R package “JASPAR2014”. There are total 593 PFMs. We show the logo plot firstly and then the negative plot. Also, the corresponding IUPAC code is given as a reference. See <a href="http://www.bioinformatics.org/sms/iupac.html" class="uri">http://www.bioinformatics.org/sms/iupac.html</a> for the IUPAC code.</p>
<pre class="r"><code>#extract all pfms
opts = list()
PFMlist = getMatrixSet(JASPAR2014,opts)
opts2 = list()
opts2[[&quot;type&quot;]] = &quot;SELEX&quot;
#get the PFMs from SELEX
PFMselex = getMatrixSet(JASPAR2014, opts2)
#get the PFMs from &#39;CHIP&#39;
opts3 = list()
opts3[[&#39;type&#39;]] = &#39;ChIP-seq&#39;
PFMchip = getMatrixSet(JASPAR2014,opts3)

#methods in JASPAR2014
mtd=c()
for(i in 1:593){mtd[i]=ifelse(is.null(PFMlist[[i]]@tags$type),&#39;&#39;,PFMlist[[i]]@tags$type)}
unique(mtd)</code></pre>
<pre><code> [1] &quot;SELEX&quot;                      &quot;COMPILED&quot;                  
 [3] &quot;&quot;                           &quot;ChiP-seq&quot;                  
 [5] &quot;Chip-seq&quot;                   &quot;High-throughput SELEX SAGE&quot;
 [7] &quot;bacterial 1-hybrid&quot;         &quot;DNaseI footprinting&quot;       
 [9] &quot;EMSA&quot;                       &quot;PBM&quot;                       
[11] &quot;PBM, CSA and/or DIP-chip&quot;   &quot;ChIP-on-chip&quot;              
[13] &quot;ChIP-seq&quot;                   &quot;ChIP-chip&quot;                 
[15] &quot;ChIP-Seq&quot;                  </code></pre>
<pre class="r"><code>#A function to get the pwm for logomaker()
scale1=function(x){return(x/sum(x))}
pwm = function(pfm,bg=0.25){
  pfm=as.matrix(pfm)
  pseudo=sqrt(dim(pfm)[2])/4
  pwm=apply((pfm+pseudo),2,scale1)
  rownames(pwm)=c(&#39;A&#39;,&#39;C&#39;,&#39;G&#39;,&#39;T&#39;)
  colnames(pwm)=1:dim(pfm)[2]
  return(pwm)
}

# A function to get the sequence motif.
GetIUPACSequence = function(pwm, prob = 0.25) {
  iupac.table =
    c(&quot;.&quot;, &quot;A&quot;, &quot;C&quot;, &quot;M&quot;, &quot;G&quot;, &quot;R&quot;, &quot;S&quot;, &quot;V&quot;, &quot;T&quot;, &quot;W&quot;, &quot;Y&quot;, &quot;H&quot;, &quot;K&quot;, &quot;D&quot;, &quot;B&quot;, &quot;N&quot;)
  iupac.value = t((pwm &gt;= prob)) %*% c(1, 2, 4, 8) + 1
  return(paste(iupac.table[iupac.value], collapse = &quot;&quot;))
}

color_profile = list(&quot;type&quot; = &quot;per_row&quot;, 
                     &quot;col&quot; = RColorBrewer::brewer.pal(4,name =&quot;Spectral&quot;))</code></pre>
<div id="selex" class="section level3">
<h3>SELEX</h3>
<p>This section shows the logo plots of the motifs from HT-SELEX.</p>
<pre class="r"><code>for(i in c(61)){
  pfm=as.matrix(PFMselex[[i]])
  print(GetIUPACSequence(pwm(pfm)))
  logomaker(pwm(pfm),xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1,
            pop_name=paste(&#39;Logo plot SELEX&#39;))
  nlogomaker(pwm(pfm),logoheight = &#39;ic&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, IC&#39;),ylimit = 2.1)
  nlogomaker(pwm(pfm),logoheight = &#39;log&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, log&#39;),ylimit = 2.1)
  nlogomaker(pwm(pfm),logoheight = &#39;log_odds&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, log-odds&#39;),ylimit = 2.1)
}</code></pre>
<pre><code>[1] &quot;GGGRATTTCC&quot;</code></pre>
<p><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="chip-seq" class="section level3">
<h3>CHIP-seq</h3>
<p>This section shows the logo plots of the motifs from CHIP-seq.</p>
<pre class="r"><code>for(i in c(121)){
  pfm=as.matrix(PFMchip[[i]])
  print(GetIUPACSequence(pwm(pfm)))
  logomaker(pwm(pfm),xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1,
            pop_name=paste(&#39;Logo plot ChIP-seq&#39;))
  nlogomaker(pwm(pfm),logoheight = &#39;ic&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, IC&#39;),ylimit = 2.1)
  nlogomaker(pwm(pfm),logoheight = &#39;log&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, log&#39;),ylimit = 2.1)
  nlogomaker(pwm(pfm),logoheight = &#39;log_odds&#39;,xlab = &#39;position&#39;,color_profile = color_profile,frame_width = 1, pop_name=paste(&#39;Neg Logo plot, log-odds&#39;),ylimit = 2.1)
}</code></pre>
<pre><code>[1] &quot;RGAACAVAATGTHCY&quot;</code></pre>
<p><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-4-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/neg_logo_plot.Rmd/unnamed-chunk-4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>In the examples above, there are depletions of bases shown in the negative logo plot. It may be considered that the transcription factor would find both enrichment and depletion to decide whether to bind. For example, in the last plot, the depletion of T at position 7 probably means the TF â€˜dislikeâ€™ T at that position. So here to understand the mechanism of transcription factor seems vital. Trying to tell whether the transcription factor would bind to a specific site, logo plot may be more straightforward since looking at the core motif would probably enough to answer yes or no. But negative logo plot may help understand how the TF works.</p>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<!-- Insert the session information into the document -->
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.0 (2017-04-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 15063)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
[1] Logolas_1.1.2       TFBSTools_1.14.0    JASPAR2014_1.12.0  
[4] Biostrings_2.43.8   XVector_0.15.2      IRanges_2.9.19     
[7] S4Vectors_0.13.17   BiocGenerics_0.22.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.11                lattice_0.20-35            
 [3] GO.db_3.4.1                 png_0.1-7                  
 [5] Rsamtools_1.27.16           gtools_3.5.0               
 [7] rprojroot_1.2               digest_0.6.12              
 [9] R6_2.2.0                    GenomeInfoDb_1.12.0        
[11] plyr_1.8.4                  backports_1.0.5            
[13] RSQLite_1.1-2               evaluate_0.10              
[15] httr_1.2.1                  ggplot2_2.2.1              
[17] zlibbioc_1.21.0             lazyeval_0.2.0             
[19] annotate_1.54.0             R.utils_2.5.0              
[21] R.oo_1.21.0                 Matrix_1.2-9               
[23] rmarkdown_1.6               splines_3.4.0              
[25] BiocParallel_1.10.0         readr_1.1.1                
[27] stringr_1.2.0               CNEr_1.12.0                
[29] RCurl_1.95-4.8              munsell_0.4.3              
[31] DelayedArray_0.2.0          compiler_3.4.0             
[33] rtracklayer_1.35.12         seqLogo_1.42.0             
[35] DirichletMultinomial_1.18.0 htmltools_0.3.5            
[37] KEGGREST_1.16.0             SummarizedExperiment_1.6.0 
[39] tibble_1.3.0                GenomeInfoDbData_0.99.0    
[41] matrixStats_0.52.2          XML_3.98-1.6               
[43] TFMPvalue_0.0.6             GenomicAlignments_1.11.12  
[45] bitops_1.0-6                R.methodsS3_1.7.1          
[47] grid_3.4.0                  xtable_1.8-2               
[49] gtable_0.2.0                DBI_0.6-1                  
[51] git2r_0.18.0                magrittr_1.5               
[53] scales_0.4.1                stringi_1.1.5              
[55] reshape2_1.4.2              RColorBrewer_1.1-2         
[57] tools_3.4.0                 BSgenome_1.44.0            
[59] Biobase_2.35.1              poweRlaw_0.70.0            
[61] hms_0.3                     yaml_2.1.14                
[63] AnnotationDbi_1.38.1        colorspace_1.3-2           
[65] GenomicRanges_1.27.23       caTools_1.17.1             
[67] memoise_1.1.0               VGAM_1.0-3                 
[69] knitr_1.15.1               </code></pre>
</div>
</div>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
